internal/controller/streams.go

```diff
@@ -271,8 +280,39 @@ func (sm *streamManager) StreamFalcoNetworkFlows(ctx context.Context) error {
 	}
 }
 
-// connectAndStreamCiliumNetworkFlows creates networkFlowsStream client and begins the streaming of network flows.
-func connectAndStreamCiliumNetworkFlows(logger *zap.Logger, sm *streamManager) error {
+// StreamKeepalives loops infinitely as long as the keepalives are working. We
+// halt when the 'keepaliveDone' channel is written to, or when sending a
+// keepalive ping fails.
+//
+// This should be run inside a goroutine in every `connectAndStream*` function
+func (sm *streamManager) StreamKeepalives(
+	freq time.Duration,
+	sendKeepalive func(*streamManager) error,
+	keepaliveDone chan struct{},
+) {
```

Comment:
1. Just pass a Context instead of reinventing the wheel with a channel.
1. Return an `error`?
```suggestion
func (sm *streamManager) StreamKeepalives(
	ctx context.Context,
	freq time.Duration,
	sendKeepalive func(*streamManager) error,
	keepaliveDone chan struct{},
) error {
```

Reply:
I don't think it makes sense to return an error as I intend this function to be run as a goroutine. So no one will ever hear the returned error.

Reply:
Using a `ctx` instead of a `keepaliveDone` channel, ok I will work on this.

Reply:
> I don't think it makes sense to return an error as I intend this function to be run as a goroutine. So no one will ever hear the returned error.

But - ok why throw out information. I will include this